### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by rk-pdx**
**What changes I made** and **Why I made these changes**

For exercise one, I simply used `else if` statements to catch all the necessary routes as per the instructions. Inside these statements, I wrote the appropriate server response, examining certain data in the header if necessary.

For exercise two, I converted the url to a string, manipulated it, and extracted the query parameters. Next, I created an inline HTML table (with a simple border style) and dynamically populated the rows with the query parameters. Instead of using inline CSS, I tried to serve a static CSS file for my inline HTML table, but was not entirely sure how to do so; if someone knows how to do this, please let me know.

For exercise three, if the url was `/`, I created a link that the user could click to access the form. I placed the static HTML code inside an external file called `form.html` and served it using `fs.createReadStream()`. If the user clicked the "submit" button on the form, it resulted in a POST request and navigated to the `/submit,` URL, which then gathered all the incoming data. Upon completion of gathering the data, in `request.on('end', ...)`, I manipulate it slightly and write it in HTML format.

**What feedback I am looking for**

As I mentioned above, I was a bit unsure of how to serve a CSS file for my inline-HTML. If anyone has ideas, I would like to know. As always, I would like to receive feedback on the style, efficiency and readability of my code, and whether I could have structured some of my logic differently. 

Thank you!

### CODE REVIEW

File: hw4/02-sessions.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const express = require('express');
 const session = require('express-session');
 const app = express();
-const port = process.env.PORT || 5000;
+const port = process.env.PORT || 5001;
```
Comment (smah29) [inline hw4/02-sessions.js:4]: you can merge upstream caterina changes in main and merge your main in this branch so that unnecessary changes are not reflected.

File: hw3/01-routing.js [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
+    `cache`,
+    `cookie`,
+    `check-cookies`,
+    `other`,
   ];
 
+
   let getRoutes = () => {
-    let result = '';
+    let result = ``;
```
Comment (smah29) [inline hw3/01-routing.js:26]: I think back ticks formatting is not required for strings

**Review by smah29 — COMMENTED** [review]
Left some comments, please see them, otherwise well written code.

File: hw3/01-routing.js (hunk 2 above)
Comment (rk-pdx) [reply hw3/01-routing.js:26]: That's correct; the back ticks are me just trying to get into the habit of using them everywhere.

File: hw4/02-sessions.js (hunk 1 above)
Comment (rk-pdx) [reply hw4/02-sessions.js:4]: Thank you!

**Comment by gleason9113** [comment]
It looks good, and I learned a couple of things that would have made my life easier.  The only thing that I see to note is that your form.js doesn't return all the values in the response.  From the example in the homework, the Comments field should display comments if present, or an n/a if not.  The newsletter response should also be shown.
